### PR TITLE
Add custom error messages for uischema validation

### PR DESCRIPTION
**Pull request by AhmetTanakol**
Added ajv custom error messages

### CODE REVIEW

**Review by edgarmueller — CHANGES_REQUESTED** [review]
Can we add or modify a test that explicitly checks for the changed error messages?

**Comment by AhmetTanakol** [comment]
Updated

File: src/models/ui-metaschema.ts [hunk 1]
```diff
@@ -71,10 +73,23 @@ export const uiMetaSchema: JsonSchema7 = {
...
-      ]
+      'required': ['type', 'scope'],
+      'additionalProperties': false,
+      'errorMessage': {
+        'properties': {
+          'type': 'type should be equal to one of the allowed values',
+          'scope': 'Control scope should match pattern "^#\\/properties\\/{1}"',
+          'suggestion': 'Control suggestion should be array',
+          'options': 'Control options should be object',
+          'label': 'Control label should be string,boolean or label object'
```
Comment (edgarmueller) [inline src/models/ui-metaschema.ts]: Add whitespace after comma

File: src/models/ui-metaschema.ts [hunk 2]
```diff
@@ -71,10 +73,23 @@ export const uiMetaSchema: JsonSchema7 = {
...
+        },
+        'required': {
+          'scope': 'Control should have an object property "scope"',
+          'type': 'Control should have an object property "type"'
```
Comment (edgarmueller) [inline src/models/ui-metaschema.ts]: Isn't type a string -> remove 'object'?

File: src/models/ui-metaschema.ts [hunk 3]
```diff
@@ -71,10 +73,23 @@ export const uiMetaSchema: JsonSchema7 = {
...
+        },
+        'additionalProperties': 'Control should not have properties ' +
+                                'other than type,label,scope,options,suggestion and rule'
```
Comment (edgarmueller) [inline src/models/ui-metaschema.ts]: Add whitespace after each comma

File: src/models/ui-metaschema.ts [hunk 4]
```diff
@@ -244,5 +304,18 @@ export const uiMetaSchema: JsonSchema7 = {
...
+  ],
+  'additionalProperties': false,
+  'errorMessage': {
+    'properties': {
+      'type': 'Root type should be equal to one of the allowed values',
+      'label': 'Root label should be string'
+    },
+    'required': {
+      'elements': 'Root should have an array property "elements"',
+      'type': 'Root should have an object property "type"'
```
Comment (edgarmueller) [inline src/models/ui-metaschema.ts]: Should be of type string

File: src/models/ui-metaschema.ts [hunk 5]
```diff
@@ -244,5 +304,18 @@ export const uiMetaSchema: JsonSchema7 = {
...
+    },
+    'additionalProperties': 'Root should not have properties ' +
+                            'other than type,elements,label and rule'
```
Comment (edgarmueller) [inline src/models/ui-metaschema.ts]: Whitespace after each comma

**Review by edgarmueller — CHANGES_REQUESTED** [review]
Error messages regarding type seem to be wrong.

**Review by edgarmueller — APPROVED** [review]

**Pull request merged**